examples/nlp/token_classification/punctuate_capitalize_infer.py

```diff
@@ -0,0 +1,739 @@
+        acc_prob: numpy array of shape
+            ``[number_of_words_for_which_probabilities_are_accumulated - number_of_probs_to_move, number_of_labels]``
+    """
+    if number_of_probs_to_move > acc_prob.shape[0]:
+        raise ValueError(
+            f"Not enough accumulated probabilities. Number_of_probs_to_move={number_of_probs_to_move} "
+            f"acc_prob.shape={acc_prob.shape}"
+        )
+    if number_of_probs_to_move > 0:
+        pred = pred + list(np.argmax(acc_prob[:number_of_probs_to_move], axis=-1))
+    acc_prob = acc_prob[number_of_probs_to_move:]
+    return pred, acc_prob
+
+
+def _update_accumulated_probabilities(acc_prob: np.ndarray, update: np.ndarray) -> np.ndarray:
+    """
+    Args:
+        acc_prob: numpy array of shape ``[A, L]``
+        update: numpy array of shape ``[A + N, L]``
+    Returns:
+        numpy array of shape ``[A + N, L]``
+    """
+    acc_prob *= update[: acc_prob.shape[0]]
+    acc_prob = np.concatenate([acc_prob, update[acc_prob.shape[0] :]], axis=0)
+    return acc_prob
+
+
+def _remove_margins(tensor, margin_size, keep_left, keep_right):
+    if not keep_left:
+        tensor = tensor[margin_size + 1 :]  # remove left margin and CLS token
+    if not keep_right:
+        tensor = tensor[: tensor.shape[0] - margin_size - 1]  # remove right margin and SEP token
+    return tensor
+
+
+def apply_punct_capit_predictions(cfg: DictConfig, query: str, punct_preds: List[int], capit_preds: List[int]) -> str:
+    """
+    Restores punctuation and capitalization in ``query``.
+    Args:
+        cfg: config for ``nemo.collections.nlp.models.PunctuationCapitalizationModel``
+        query: a string without punctuation and capitalization
+        punct_preds: ids of predicted punctuation labels
+        capit_preds: ids of predicted capitalization labels
+    Returns:
+        a query with restored punctuation and capitalization
+    """
+    query = query.strip().split()
+    assert len(query) == len(
+        punct_preds
+    ), f"len(query)={len(query)} len(punct_preds)={len(punct_preds)}, query[:30]={query[:30]}"
+    assert len(query) == len(
+        capit_preds
+    ), f"len(query)={len(query)} len(capit_preds)={len(capit_preds)}, query[:30]={query[:30]}"
+    punct_ids_to_labels = {v: k for k, v in cfg.punct_label_ids.items()}
+    capit_ids_to_labels = {v: k for k, v in cfg.capit_label_ids.items()}
+    query_with_punct_and_capit = ''
+    for j, word in enumerate(query):
+        punct_label = punct_ids_to_labels[punct_preds[j]]
+        capit_label = capit_ids_to_labels[capit_preds[j]]
+
+        if capit_label != cfg.dataset.pad_label:
+            word = word.capitalize()
+        query_with_punct_and_capit += word
+        if punct_label != cfg.dataset.pad_label:
+            query_with_punct_and_capit += punct_label
+        query_with_punct_and_capit += ' '
+    return query_with_punct_and_capit[:-1]
+
+
+def _transform_logit_to_prob_and_remove_margins_and_extract_word_probs(
```

Comment:
move to the model class or create a utils file inside https://github.com/NVIDIA/NeMo/tree/main/nemo/collections/nlp/models/token_classification